src/components/wms/AnimatedStreamlineMapboxLayer.vue

```diff
@@ -0,0 +1,130 @@
+
+interface Props {
+  layerOptions?: MapboxLayerOptions
+  streamlineOptions?: StreamlineLayerOptionsFews
+}
+const props = defineProps<Props>()
+const emit = defineEmits(['doubleclick'])
+
+const { map } = useMap()
+
+const layerId = 'streamlines'
+let layer: WMSStreamlineLayer | null = null
+
+onMounted(() => {
+  if (!props.layerOptions || !props.streamlineOptions) return
+  const options = mergeOptions(props.layerOptions, props.streamlineOptions)
+
+  // Create and add layer.
+  layer = new WMSStreamlineLayer(layerId, options)
+  map.value.addLayer(layer, 'boundary_country_outline')
+  map.value.on('dblclick', onDoubleClick)
+
+  // Make sure we are at the appropriate time, elevation and color scale range
+  // after the visualiser has been initialised.
+  layer.on('load', () => {
+    if (!layer || !props.layerOptions) return
+    layer.updateOptions(
+      options,
+      props.layerOptions.time,
+      props.layerOptions.elevation ?? undefined,
+      getColorScaleRangeFromString(props.layerOptions.colorScaleRange),
+    )
+  })
+})
+
+onUnmounted(() => {
+  map.value.removeLayer(layerId)
+  map.value.off('dblclick', onDoubleClick)
+})
+
+// Update the visualiser when the layer is changed.
+watch(
+  () => props.layerOptions?.name,
+  async () => {
+    if (!layer || !props.layerOptions) return
+
+    const options = mergeOptions(props.layerOptions, props.streamlineOptions)
+    await layer.updateOptions(
+      options,
+      props.layerOptions.time,
+      props.layerOptions.elevation ?? undefined,
+      getColorScaleRangeFromString(props.layerOptions.colorScaleRange),
+    )
+  },
+)
+
+// Update the velocity field when the time, elevation or color scale range is
+// changed.
+watch(
+  () => props.layerOptions?.time,
+  (time) => {
+    if (!layer || !time) return
+    layer.setTime(time)
+  },
+)
+watch(
+  () => props.layerOptions?.elevation,
+  (elevation) => {
+    if (!layer) return
+    layer.setElevation(elevation ?? null)
+  },
+)
+watch(
+  () => props.layerOptions?.colorScaleRange,
+  (colorScaleRange) => {
+    if (!layer) return
+    layer.setColorScaleRange(
+      getColorScaleRangeFromString(colorScaleRange) ?? null,
+    )
+  },
+)
+
+function onDoubleClick(event: MapLayerMouseEvent | MapLayerTouchEvent) {
+  emit('doubleclick', event)
+}
+
+function mergeOptions(
+  layerOptions: MapboxLayerOptions,
+  streamlineOptions: StreamlineLayerOptionsFews,
+): WMSStreamlineLayerOptions {
+  const baseUrl = configManager.get('VITE_FEWS_WEBSERVICES_URL')
+  const baseUrlWms = `${baseUrl}/wms`
+  return {
+    baseUrl: baseUrlWms,
+    layer: layerOptions.name,
+    streamlineStyle: StreamlineStyle.ColoredParticles,
+    numParticles: streamlineOptions?.numberOfParticles ?? 1000,
+    particleSize: streamlineOptions?.particleSize ?? 3,
+    speedFactor: streamlineOptions?.speedFactor ?? 0.2,
+    fadeAmountPerSecond: streamlineOptions?.fadeAmount ?? 0.1,
+    particleColor: `#${streamlineOptions?.particleColor}`,
```

Comment:
Wouldn't it be better to set the '#' in the `particleColor`? That way, a `particleColor` of `rgb(r,g,b)` or `'black'` would also be possible.

Reply:
Yes it would be, but this is a back-end problem...